sympy/sets/handlers/comparison.py

```diff
@@ -57,3 +51,29 @@ def _eval_is_eq(lhs, rhs): # noqa: F811
 @dispatch(Set, Set) # type:ignore
 def _eval_is_eq(lhs, rhs): # noqa: F811
     return None
+
+
+@dispatch(ProductSet, FiniteSet) # type:ignore
+def _eval_is_eq(lhs, rhs): # noqa: F811
+    if len(lhs.sets) == len(rhs):
+        def all_in_both():
+            s_set = set(lhs.args)
+            o_set = set(rhs.args)
+            yield fuzzy_and(lhs._contains(e) for e in o_set - s_set)
+            yield fuzzy_and(rhs._contains(e) for e in s_set - o_set)
+
+        return tfn[fuzzy_and(all_in_both())]
+    return tfn[False]
+
+
+@dispatch(FiniteSet, ProductSet) # type:ignore
```

Comment:
I don't think it's necessary to define the dispatcher both ways round. The `is_eq` function will try flipping the arguments
https://github.com/sympy/sympy/blob/0bde5c951b52341e745d5410dba1a9a75d7acebf/sympy/core/relational.py#L1344-L1351

Even if it was necessary to do this both ways round it would clearly be better to use a function rather than duplicate code like this.